### PR TITLE
Adds login for demo to Login and Signup pages

### LINKED ISSUE #37: FE: Demo User
This ticket does not have a mock-up so please add in the button in the Login and Sign up pages and make sure the design is consistent with the overall look and feel.

- Create a "demo user" functionality that will be used later to allow recruiters and managers to easily test the app's features without having to sign up
- Add a button to allow for signing in as a demo user
- This is effectively a login button with a hardcoded username and password
- Show button both on signup and login pages
- In production you should manually create a user and some test data that would best show off your project. Do this in development right now to test this feature

### PR DESCRIPTION
**Pull request by igor-tanv**
### What this PR does (required):
- creates a Login for a demo user

### Screenshots / Videos (front-end only):
- Screen Shot 2022-01-16 at 9 55 52 PM
- 
### Any information needed to test this feature (required):
- Click 'Login As Demo User'

### Any issues with the current functionality (optional):
- This PR has one error: 
````
Proxy error: Could not proxy request /auth/demo from localhost:3000 to http://localhost:3001.
````
NOTE: 'auth/demo' endpoint works successfully when using Postman and all other endpoints work as expected. Please advise.

### CODE REVIEW

**Comment by sundayezeilo** [comment]
Please link your ticket before we can continue with the review

File: client/src/helpers/APICalls/loginDemoUser.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { AuthApiData } from '../../interface/AuthApiData';
+import { FetchOptions } from '../../interface/FetchOptions';
+
+const loginDemoUser = async (): Promise<AuthApiData> => {
+  const fetchOptions: FetchOptions = {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify({}),
```
Comment (moffatethan) [inline client/src/helpers/APICalls/loginDemoUser.ts:8]: This can be removed

**Review by moffatethan — APPROVED** [review]
Nice work! Just one quick change then you can merge in

File: client/src/helpers/APICalls/loginDemoUser.ts (hunk 1 above)
Comment (igor-tanv) [reply client/src/helpers/APICalls/loginDemoUser.ts:8]: Ethan,

I'm not sure if you saw my PR comment about the proxy error so I'll past it again here:
````
Proxy error: Could not proxy request /auth/demo from localhost:3000 to http://localhost:3001.
````
This error occurs when I click the 'Login As Demo User' button. 

Removing fetchOptions does not resolve this error. Please help me understand the problem. All other routes are working.

Comment (moffatethan) [reply client/src/helpers/APICalls/loginDemoUser.ts:8]: Super odd fix - in the package.json the proxy should be http://localhost:3001**/** it's currently just http://localhost:3001

Comment (moffatethan) [reply client/src/helpers/APICalls/loginDemoUser.ts:8]: Everything else looks fine to me & works after doing that fix in the proxy. Not a thorough fix but it gets it working - if we wanted to get a better fix in we would turn to the underlying webpack configuration doe the client dev server proxy settings.

**Review by moffatethan — APPROVED** [review]
Please take a look at the comment regarding the package.json hotfix for this issue you're having. Tested and works fine on my end, good work

**Pull request merged**